src/test/java/com/glencoesoftware/omero/zarr/ZarrPixelBufferTest.java

```diff
@@ -786,4 +821,264 @@ public void testDownsampledZ() throws IOException {
+            createPixelBuffer(pixels, testZarrPath.resolve("0"), testZarr.getSizeX(), testZarr.getSizeY())) {
+                Map<Axis, Integer> axes = zpbuf.getAxesOrder();
+                Assert.assertEquals(4 - order.indexOf("Z"), axes.get(Axis.Z).intValue());
+                Assert.assertEquals(4 - order.indexOf("T"), axes.get(Axis.T).intValue());
+                Assert.assertEquals(4 - order.indexOf("C"), axes.get(Axis.C).intValue());
+                Assert.assertEquals(4 - order.indexOf("Y"), axes.get(Axis.Y).intValue());
+                Assert.assertEquals(4 - order.indexOf("X"), axes.get(Axis.X).intValue());
+                Assert.assertEquals(testZarr.getSizeT(), zpbuf.getSizeT());
+                Assert.assertEquals(testZarr.getSizeC(), zpbuf.getSizeC());
+                Assert.assertEquals(testZarr.getSizeZ(), zpbuf.getSizeZ());
+                Assert.assertEquals(testZarr.getSizeY(), zpbuf.getSizeY());
+                Assert.assertEquals(testZarr.getSizeX(), zpbuf.getSizeX());
+        }
+    }
+
+    private void testDimensions(int sizeX, int sizeY, int sizeZ, int sizeC, int sizeT) throws IOException, InvalidRangeException {
+        int textX = 10;
+        int textY = 10;
+        
+        String order = DimensionOrder.VALUE_XYZCT;
+        if (sizeT == 0) {
+            order.replace("T", "");
+        }
+        if (sizeZ == 0) {
+            order.replace("Z", "");
+        }
+        if (sizeC == 0) {
+            order.replace("C", "");
+        }
+        
+        Path testZarrPath = tmpDir.getRoot().toPath().resolve("test.zarr");
+        TestZarr testZarr = new TestZarr()
+            .setPath(testZarrPath)
+            .setOverwrite(true)
+            .setSizeX(sizeX)
+            .setSizeY(sizeY)
+            .setSizeZ(sizeZ)
+            .setSizeT(sizeT)
+            .setSizeC(sizeC)
+            .setOrder(new StringBuilder(order).reverse().toString())
+            .setTextX(textX)
+            .setTextY(textY)
+            .init()
+            .createImage()
+            .createMetadata();
+
+        int pixZ = sizeZ > 0 ? sizeZ : 1;
+        int pixC = sizeC > 0 ? sizeC : 1;
+        int pixT = sizeT > 0 ? sizeT : 1;
+        
+        Pixels pixels = new Pixels(
+            null, new PixelsType(PixelsType.VALUE_INT32), 
+            sizeX, sizeY, pixZ, pixC, pixT, "", new DimensionOrder(DimensionOrder.VALUE_XYZCT));
+        
+        int expectedTests = pixC * pixT * pixZ;
+        int testCount = 0;
+        try (ZarrPixelBuffer zpbuf = createPixelBuffer(pixels, testZarrPath.resolve("0"), sizeX, sizeY)) {
+            for (int t = 0; t <= sizeT; t++) {
+                if (t == sizeT && sizeT > 0)
+                    break;
+                for (int z = 0; z <= sizeZ; z++) {
+                    if (z == sizeZ && sizeZ > 0)
+                        break;
+                    for (int c = 0; c <= sizeC; c++) {
+                        if (c == sizeC && sizeC > 0)
+                            break;
```

Comment:
As in `TestZarr` lines 178-180, this would be easier to follow if using something like `for (int t = 0; t < Math.max(sizeT, 1); t++)`.